packages/typespec-ts/src/index.ts

```diff
@@ -332,36 +343,34 @@ export async function $onEmit(context: EmitContext) {
         rootIndexFile
       );
     }
+    console.timeEnd("onEmit: emit source files");
     // Enable modular sample generation when explicitly set to true or MPG
     if (emitterOptions.generateSample === true) {
+      console.time("onEmit: emit samples");
       const samples = emitSamples(dpgContext);
+      console.timeEnd("onEmit: emit samples");
       // Refine the rlc sample generation logic
       // TODO: remember to remove this out when RLC is splitted from Modular
       if (samples.length > 0) {
         dpgContext.rlcOptions!.generateSample = true;
       }
     }
 
+    console.time("onEmit: resolve references");
     binder.resolveAllReferences(modularSourcesRoot);
+    console.timeEnd("onEmit: resolve references");
+
+    console.time("onEmit: generate files");
 
     for (const file of project.getSourceFiles()) {
-      file.fixMissingImports(
```

Comment:
Can i understand the main improvement is because of importing operations in ts-morph is quite time-comsuming?

So we switched all imports to binder which fasten a lot?

Reply:
no, ts-morph itself is not that slow, it's the tricky thing we did here to mitigate the half binder half ts-morph situation that causes things slow.